### PR TITLE
Rename Phase.symmetry to .point_group, change cm/IO accordingly

### PR DESCRIPTION
**Pull request by hakonanes**
Signed-off-by: Håkon Wiik Ånes <hwaanes@gmail.com>

Renames `Phase.symmetry` to `Phase.point_group`, in anticipation of adding `Phase.space_group`, from `diffpy.structure.spacegroupmod.SpaceGroup`, as discussed in #87. Accompanying changes are made to the `PhaseList` and `CrystalMap` classes, as well as all IO plugins.

The `CrystalMap`, `Phase` and `PhaseList` representations now look like this:

```python
>>> cm
Phase  Orientations       Name  Point group       Color
    1  5657 (48.4%)  austenite          432    tab:blue
    2  6043 (51.6%)    ferrite          432  tab:orange
Properties: iq, dp
Scan unit: um
>>> cm.phases
Id       Name  Point group       Color
 1  austenite          432    tab:blue
 2    ferrite          432  tab:orange
>>> cm.phases["austenite"]
<name: austenite. point group: 432. color: tab:blue>
>>> cm.phases["austenite"].point_group
Symmetry (24,) 432
[[ 1.      0.      0.      0.    ]
[...]  # Many rotations
 [ 0.5    -0.5     0.5     0.5   ]]
```

Initializing `CrystalMap`, `Phase` and `PhaseList` is now done by passing a `point_group` argument instead of `symmetry`.

Issues resulting from this PR:
* Might the naming of `point_group` compared to `symmetry` be confusing for the user? I actually think it might be a better naming and help the user's understanding.  
* The orix HDF5 plugin now writes the phase point group to an HDF5 dataset with name `point_group` and not `symmetry`, with similar changes when reading. So, if this PR is merged into master, reading a phase HDF5 group  from a file written with the current writer won't work... unless we check for this in the reader. Doing this is easy. However, it raises the question of how many such small changes we should accomodate.

### CODE REVIEW

**Comment by pc494** [comment]
_" The orix HDF5 plugin now writes the phase point group to an HDF5 dataset with name point_group and not symmetry, with similar changes when reading. So, if this PR is merged into master, reading a phase HDF5 group from a file written with the current writer won't work... unless we check for this in the reader. Doing this is easy. However, it raises the question of how many such small changes we should accomodate."_

I think the reader should be stable within minor versions (without any clever additions). If the hdf5 contains the version we could then support a finite number of versions at any given time. Say nine months worth?

Worst case, you've saved something with v0.3 and that version is no longer supported (say we've got to v0.6), your workaround (tedious) is that you have to load with 0.4 save to 0.4 load with 0.6 and save again or similar?

The problem gets smaller if we try to make all the breaking changes one release cycle at a time?

tl;dr ~ happy for this change, and would be inclined to support v0.3 in this instance

**Comment by hakonanes** [comment]
> I think the reader should be stable within minor versions (without any clever additions). If the hdf5 contains the version we could then support a finite number of versions at any given time. Say nine months worth?

The orix version is always written to the top HDF5 group "/" in the `version` dataset. Nine months... two minor versions? Sounds good to me.

> Worst case, you've saved something with v0.3 and that version is no longer supported (say we've got to v0.6), your workaround (tedious) is that you have to load with 0.4 save to 0.4 load with 0.6 and save again or similar?

Should have mentioned this above... but the simplest solution is to open the file in HDFView (or with h5py) and rename the dataset. I do this all the time with EMsoft HDF5 files when some dataset names or values are inconsistent.

> The problem gets smaller if we try to make all the breaking changes one release cycle at a time?

Every time we release a major version? Agreed, we should adhere to semver.

> tl;dr ~ happy for this change, and would be inclined to support v0.3 in this instance

Okidoki, will add a check for the different dataset names.

**Comment by hakonanes** [comment]
Instead of checking which orix version the orix HDF5 file was written with, I just used

```python
try:
    point_group = dictionary["point_group"]
except KeyError:
    point_group = dictionary["symmetry"]
```

Also added a TODO comment to remove the check in orix v0.6.

**Comment by pc494** [comment]
Is this good for review now?

**Comment by hakonanes** [comment]
yup, ready for review

File: orix/crystal_map/crystal_map.py [hunk 1]
```diff
@@ -610,26 +610,28 @@ def __repr__(self):
...
         p_sizes = [np.where(phase_ids == i)[0].size for i in unique_phases]
         id_len = 5
-        ori_len = max(max([len(str(p_size)) for p_size in p_sizes]) + 9, 13)
-        name_len = max(max([len(n) for n in names]), 5)
-        sym_len = max(max([len(sn) for sn in symmetry_names]), 8)
-        col_len = max(max([len(i) for i in phases.colors]), 6)
+        ori_len = max(max([len(str(p_size)) for p_size in p_sizes]) + 8, 12)
+        name_len = max(max([len(n) for n in names]), 4)
+        pg_len = max(max([len(sn) for sn in point_group_names]), 11)
+        col_len = max(max([len(i) for i in phases.colors]), 5)
```
Comment (pc494) [inline orix/crystal_map/crystal_map.py:624]: is the changing numbers here important?

File: orix/io/plugins/orix_hdf5.py [hunk 2]
```diff
@@ -164,6 +164,8 @@ def dict2phase(dictionary):
     ----------
     dictionary : dict
         Dictionary with phase information.
+    version : list, optional
```
Comment (pc494) [inline orix/io/plugins/orix_hdf5.py:167]: this can be deleted now?

File: orix/io/plugins/orix_hdf5.py [hunk 3]
```diff
@@ -172,13 +174,17 @@ def dict2phase(dictionary):
...
     structure.title = dictionary["name"]
-    symmetry = dictionary["symmetry"]
-    if symmetry == "None":
-        symmetry = None
+    # TODO: Remove this check in v0.6.0
+    try:
+        point_group = dictionary["point_group"]
+    except KeyError:  # v0.3.0
+        point_group = dictionary["symmetry"]
+    if point_group == "None":
```
Comment (pc494) [inline orix/io/plugins/orix_hdf5.py:182]: PEP 8 prefers `is None`, although it's certainly more of a convention than a necessity

File: orix/crystal_map/crystal_map.py (hunk 1 above)
Comment (hakonanes) [reply orix/crystal_map/crystal_map.py:624]: This ensures the `repr()` table looks good when changing `Symmetry` to `Point group`.

File: orix/io/plugins/orix_hdf5.py (hunk 3 above)
Comment (hakonanes) [reply orix/io/plugins/orix_hdf5.py:182]: If a phase has no point group set, it is written as a string "None" to the HDF5 file, an subsequently read as a string "None". So I'm pretty sure this check is correct.

Comment (hakonanes) [reply orix/io/plugins/orix_hdf5.py:182]: Of course, I could add a conversion from string "None" to actually `None` in the HDF5 reader. Will look into it.

Comment (pc494) [reply orix/io/plugins/orix_hdf5.py:182]: Nope, I was reading on auto-pilot, the string "None" is the way to go here. No need for further action.

File: orix/io/plugins/orix_hdf5.py (hunk 2 above)
Comment (hakonanes) [reply orix/io/plugins/orix_hdf5.py]: Thanks, deleted.

**Comment by hakonanes** [comment]
Should be good now.
